### PR TITLE
feat: add extension point for application menu items

### PR DESCRIPTION
**Pull request by JammingBen**
## Description
Adds an extension point for adding items in the top left application switcher menu.

`AppMenuItemExtension`'s can either link to an internal page (via `path`), an external page (via `url`) or have a button handler (via `handler`). Please refer to the docs for more information.

This deprecates the `applicationMenu` property of the appinfo object.

Also removes ancient menu logic from oC10 times, such as the user menu.

## Related Issue

- works towards https://github.com/owncloud/web/issues/7260

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests
- [ ] Documentation
- [ ] Maintenance (e.g. dependency updates or tooling)

### CODE REVIEW

**Comment by JammingBen** [comment]
This currently only supports remix icons. Do we need support for icon urls?

File: changelog/unreleased/enhancement-app-menu-item-extension-point [hunk 1]
```diff
@@ -0,0 +1,9 @@
+Enhancement: Application menu extension point
+
+We've added an extension point for adding items in the top left application switcher menu.
+
+`AppMenuItemExtension`'s can either link to an internal page (via `path`), an external page (via `url`) or have a button handler (via `handler`). Please refer to the docs for more information.
+
+DEPRECATION NOTICE: This deprecates the `applicationMenu` property of the appinfo object.
```
Comment (kulmann) [inline changelog/unreleased/enhancement-app-menu-item-extension-point:7]: I wanted to ask you to also add the deprecation to https://github.com/owncloud/web/issues/7338 but you already did that :D

**Review by kulmann — COMMENTED** [review]
When I do a production build there is a vue-router error when navigating into an embedded `wikipedia.com`. doesn't happen with vite 🤔  (tested together with a production build of https://github.com/owncloud/web-extensions/pull/13 in my ocis apps folder)

**Comment by JammingBen** [comment]
> When I do a production build there is a vue-router error when navigating into an embedded `wikipedia.com`. doesn't happen with vite 🤔 (tested together with a production build of [owncloud/web-extensions#13](https://github.com/owncloud/web-extensions/pull/13) in my ocis apps folder)

Huh, there was an issue with the way I've added the router component. I don't get why this didn't pop up in dev mode... anyway, now it should work (although you need to edit your server's CSP rules to allow an external site being embedded via an iFrame).

File: docs/extension-system/extension-types/app-menu-items.md [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+  priority?: number
+  url?: string
+}
+```
+
+For `id`, `type`, and `extensionPointIds`, please see [extension base section]({{< ref "../_index.md#extension-base-configuration" >}}) in the top level docs.
+
+A `handler` will result in a `<button>` element. This is necessary when an action should be performed when clicking the menu item (e.g. opening a file editor).
+
+A `path` will result in a `<a>` element that links to an internal page. That means the given path needs to exist within the application.
```
Comment (kulmann) [inline docs/extension-system/extension-types/app-menu-items.md:40]:
```suggestion
A `path` will result in an `<a>` element that links to an internal page via the vue router. That means the given path needs to exist within the application.
```

File: docs/extension-system/extension-types/app-menu-items.md [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+
+A `url` will result in a `<a>` element that links to an external page. External pages always open in a new tab or window.
```
Comment (kulmann) [inline docs/extension-system/extension-types/app-menu-items.md:42]:
```suggestion
A `url` will result in an `<a>` element that links to an external page. External pages always open in a new tab or window.
```

File: docs/extension-system/extension-types/app-menu-items.md [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+
+Since these properties are optional, the priority order is `handler` > `path` > `url`. At least one of them has to be provided when registering an extension.
```
Comment (kulmann) [inline docs/extension-system/extension-types/app-menu-items.md:44]:
```suggestion
At least one of these properties has to be provided when registering an extension. If you define more than one, the priority order is `handler` > `path` > `url`. 
```

**Review by kulmann — CHANGES_REQUESTED** [review]
Hmmm... did I overlook something or is an `isVisible` callback missing? There are apps, e.g. the `admin-settings` app, which are only visible to certain users.

**Comment by JammingBen** [comment]
> Hmmm... did I overlook something or is an isVisible callback missing? There are apps, e.g. the admin-settings app, which are only visible to certain users.

Do we need it though? Since it will be the app's duty to register menu items, it can just not register an item if the user doesn't have permissions to see it.

**Comment by kulmann** [comment]
> > Hmmm... did I overlook something or is an isVisible callback missing? There are apps, e.g. the admin-settings app, which are only visible to certain users.
> 
> Do we need it though? Since it will be the app's duty to register menu items, it can just not register an item if the user doesn't have permissions to see it.

Hm true...

**Review by kulmann — APPROVED** [review]

**Pull request merged**